src/filter.c

```diff
@@ -12,21 +12,32 @@
 #include <assert.h>
 #include <zlib.h>
 #include <getopt.h>
+#include <time.h>
 
 #define block_size 2048
 #define unsafe_block_size 4096
 
 int threshold;
+char *r1i_path = NULL, *r2i_path = NULL, *r1o_path = NULL, *r2o_path = NULL, *stats_file = NULL;
+int read_pairs_checked = 0, read_pairs_removed = 0, read_pairs_remaining = 0;
 
-static void _log(char* msg) {
-    printf("[fastq_filterer] %s\n", msg);
+
+static void timestamp() {
+
+    time_t t = time(NULL);
+    struct tm* now = localtime(&t);
+    
+    printf(
+        "[%i-%i-%i %i:%i:%i][fastq_filterer] ",
+        now->tm_year + 1900, now->tm_mon + 1, now->tm_mday, now->tm_hour, now->tm_min,  now->tm_sec
+    );
```

Comment:
Not sure why you chose to log the timestamp separately from the message.
Adding the timestamp in the _log function would have made more sense no ?